org.eclipse.lsp4e.test/src/org/eclipse/lsp4e/test/TestUtils.java

```diff
@@ -233,17 +237,66 @@ public static IEditorReference[] getEditors() {
 		return null;
 	}
 
-	public static boolean waitForCondition(int timeout_ms, BooleanSupplier condition) {
+	public static void waitForAndAssertCondition(int timeout_ms, Condition condition) {
+		assertTrue("Condition not met within expected time.",
+				waitForCondition(timeout_ms, condition));
+	}
+
+	public static void waitForAndAssertCondition(int timeout_ms, Display display, Condition condition) {
+		assertTrue("Condition not met within expected time.",
+				waitForCondition(timeout_ms, display, condition));
+	}
+
+	public static void waitForAndAssertCondition(String errorMessage, int timeout_ms, Condition condition) {
+		assertTrue(errorMessage, waitForCondition(timeout_ms, condition));
+	}
+
+	public static void waitForAndAssertCondition(String errorMessage, int timeout_ms, Display display, Condition condition) {
+		var ex = new Exception[1];
```

Comment:
why do we use an array with one position instead of just a simple variable?

Reply:
this stems from the fact that inner anonymous/lambdas instances can only access final local fields but we need to write to a field.